src/meta/src/storage/etcd_retry_client.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::iter::{Map, Take};
+use std::time::Duration;
+
+use etcd_client::{
+    DeleteOptions, DeleteResponse, Error, GetOptions, GetResponse, KvClient, PutOptions,
+    PutResponse, Txn, TxnResponse,
+};
+use tokio_retry::strategy::{jitter, FixedInterval};
+
+type Result<T> = std::result::Result<T, etcd_client::Error>;
+
+const DEFAULT_RETRY_INTERVAL: Duration = Duration::from_millis(500);
+const DEFAULT_RETRY_MAX_ATTEMPTS: usize = 20;
```

Comment:
We're not sure whether these parameters are good. May investigate later in further tests.

Reply:
How about using [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) for all retries? May do it later.

Reply:
Here is a more clear description: https://docs.aws.amazon.com/general/latest/gr/api-retries.html